### PR TITLE
Setup Redux

### LINKED ISSUE #7: Setup redux

### PR DESCRIPTION
**Pull request by mohitkyadav**
Fixes #7

### CODE REVIEW

File: src/components/common/Header.js [hunk 1]
```diff
@@ -9,8 +9,8 @@ class Header extends Component {
             <div className="uk-navbar-left">
             <BrowserRouter>
                 <ul className="uk-navbar-nav">
-                    <li><Link to="/">Home</Link></li>
-                    <li><Link to="/home">Algos</Link></li>
+                  <li><Link to="/">Home</Link></li>
```
[the remark this answers is not in the document]
Comment (mohitkyadav) [reply src/components/common/Header.js]: Style guide rules are defined in .editorconfig, should I also configure eslint?

**Comment by yashLadha** [comment]
Prettier and eslint would be a good combination.

File: src/components/common/Header.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (singhpratyush) [reply src/components/common/Header.js]: Yes. Maybe in another patch.

File: src/actions/ghActions.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+import {
+  LOADING,
+  LOADED,
+  // ERROR,
+  // AUTHOR_PROFILE
+} from './types';
+
+// import isEmpty from './../utils/validate';
+
+export const loading = () => {
```
Comment (yashLadha) [inline src/actions/ghActions.js]:
```suggestion
export const loading = () => ({
     type: <any_data>,
})
```

Reduce lot of lines and looks clean :sweat_smile:

File: src/components/welcome/Header.js [hunk 3]
```diff
@@ -17,3 +31,9 @@ export default class Header extends Component {
     )
   }
 }
+
+const mapStateToProps = state => ({
+  gh: state.gh
+});
+
+export default connect(mapStateToProps, { loading, loaded })(withRouter(Header));
```
Comment (yashLadha) [inline src/components/welcome/Header.js]: create a variable for dispatch props

**Comment by yashLadha** [comment]
Also you can refer to this post to lint the code: [link](https://medium.com/technical-credit/using-prettier-with-vs-code-and-create-react-app-67c2449b9d08)

File: src/components/welcome/Header.js (hunk 3 above)
Comment (mohitkyadav) [reply src/components/welcome/Header.js]: @yashLadha this is just the header component of the welcome page, this is not even the real header, and I just put the state and actions to demonstrate the working of redux, do you still want me to do that?

Comment (yashLadha) [reply src/components/welcome/Header.js]: See it's not just to get things to work done but also make things that can be maintainable later as well. If we make a dispatch variable and later on we need to extend or some newcomer needs to extend the functionality then he gets a clear idea of where the functions are defined, and also that would be a good practice too.

**Pull request merged**